apps/total-typescript/src/pages/tutorials/[module]/[exercise]/index.tsx

```diff
@@ -1,36 +1,42 @@
 import React from 'react'
 import ExerciseTemplate from 'templates/exercise-template'
-import {GetServerSideProps} from 'next'
-import {getModule} from 'lib/tutorials'
+import {GetServerSideProps, GetStaticPaths, GetStaticProps} from 'next'
+import {getAllTutorials, getModule} from 'lib/tutorials'
 import {getExercise} from 'lib/exercises'
 
-export const getServerSideProps: GetServerSideProps = async (context) => {
+export const getStaticProps: GetStaticProps = async (context) => {
   const {params} = context
   const exerciseSlug = params?.exercise as string
 
-  const subscriber = context.req.cookies['ck_subscriber'] || null
   const module = await getModule(params?.module as string)
   const exercise = await getExercise(exerciseSlug)
 
-  if (!exercise) {
-    return {
-      notFound: true,
-    }
-  }
-
   return {
-    props: {exercise, module, subscriber},
+    props: {exercise, module},
+    revalidate: 10,
```

Comment:
this turns on isr. the `revalidate` is 10 seconds, after that it regenerates the static page

https://nextjs.org/docs/basic-features/data-fetching/incremental-static-regeneration

in theory this means you should be able to update the cdn (Sanity) and it will load with new data and recache. They recommend turning `useCdn: false` in cdn, but I don't know how that works in practice